chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -71,7 +71,11 @@ trait InstanceId {
 private[chisel3] trait HasId extends InstanceId {
   private[chisel3] def _onModuleClose: Unit = {} // scalastyle:ignore method.name
   private[chisel3] val _parent: Option[BaseModule] = Builder.currentModule
-  _parent.foreach(_.addId(this))
+  // Ignore null parents.
```

Comment:
What are the implications here of this potentially hiding bugs in the None case, since it will fail silently? Is there a way to make it fail noisily?

Reply:
Actually, the new behavior isn't a "failure". It's required so the initial `DontCare` element (created in the package object) isn't added to the parent module's list of elements. Should we check `this` and fail noisely if `this` isn't a `DontCare`? Or provide a `noParent` method which is called in cases where we expect a parent but don't find one. The default method would complain, but `DontCare` (or any other implementation object) would override it to return without complaint.

Reply:
Sure, but only for the case of those singleton objects, right? If we encountered an issue with a non-singleton object where the parent was corrupted / misinitialized / not initialized, this seems like it would produce a failure that would silent discard useful data, or one that produces an exception elsewhere.